src/main/java/apoc/merge/Merge.java

```diff
@@ -0,0 +1,59 @@
+package apoc.merge;
+
+import apoc.result.*;
+import org.neo4j.graphdb.*;
+import org.neo4j.helpers.collection.Iterators;
+import org.neo4j.procedure.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Merge {
+
+    @Context
+    public GraphDatabaseService db;
+
+    @Procedure(mode = Mode.WRITE)
+    @Description("apoc.merge.node(['Label'], {key:value, ...}, {key:value,...}) - merge node with dynamic labels")
+    public Stream<NodeResult> node(@Name("label") List<String> labelNames, @Name("identProps") Map<String, Object> identProps, @Name("props") Map<String, Object> props) {
+        String labels = String.join(":", labelNames);
+
+        Map<String, Object> params = buildParams(identProps, props);
+        String identPropsString = buildIdentPropsString(identProps);
+
+        final String cypher = "MERGE (n:" + labels + "{" + identPropsString + "}) ON CREATE SET n += $props RETURN n";
```

Comment:
I don't think empty {} will work when no id-props, isn't that an error condition to have no id-props?